src/Initializer/MemoryManager.cpp

```diff
@@ -95,7 +95,19 @@ void seissol::initializers::MemoryManager::initialize()
   // initialize global matrices
   GlobalDataInitializerOnHost::init(m_globalDataOnHost, m_memoryAllocator, MEMKIND_GLOBAL);
   if constexpr (seissol::isDeviceOn()) {
-    GlobalDataInitializerOnDevice::init(m_globalDataOnDevice, m_memoryAllocator, memory::DeviceGlobalMemory);
+    bool serialize = false;
+    if (const char* value = std::getenv("SEISSOL_SERIALIZE_NODE_DEVICE_INIT")) {
+      serialize = strcmp(value, "1") == 0;
+    }
```

Comment:
This isn't really robust as it doesn't handle incorrect values for the env flag. 
IMO there should be one check if the flag exists and one that the value is correct (and if it isn't, errors)

Reply:
That is already done by the `if`: `value == nullptr` if it does not exist. I'll make it more explicitly nonetheless.